apps/condo/domains/banking/access/BankAccount.js

```diff
@@ -22,41 +22,58 @@ async function canReadBankAccounts ({ authentication: { item: user } }) {
     if (user.deletedAt) return false
     if (user.isAdmin || user.isSupport) return {}
 
+    const conditions = {
+        OR: [],
+    }
+
+    if (user.type === STAFF) {
+        conditions.OR.push({
+            organization: queryOrganizationEmployeeFor(user.id),
+        })
+        conditions.OR.push({
+            organization: queryOrganizationEmployeeFromRelatedOrganizationFor(user.id),
+        })
+    }
+
     if (user.type === SERVICE) {
-        return {
-            OR: [{
-                integrationContext: {
+        conditions.OR.push({
+            integrationContext: {
+                deletedAt: null,
+                integration: {
                     deletedAt: null,
-                    integration: {
+                    accessRights_some: {
+                        user: { id: user.id },
                         deletedAt: null,
-                        accessRights_some: {
-                            user: { id: user.id },
-                            deletedAt: null,
-                        },
                     },
                 },
             },
-            {
-                integrationContext_is_null: true,
-            }],
+        })
+        conditions.OR.push({
+            integrationContext_is_null: true,
+        })
+    }
 
+    if (user.type === RESIDENT) {
```

Comment:
looks like heavy calculations that will be executed for every corresponding request... :(

Reply:
It will be executed for every read request, for example, one `getAll…` request.
Exactly as for `create/update` as in all modules.